Pocket Casts Watch App/NowPlayingControls.swift

```diff
@@ -131,11 +131,15 @@ struct NowPlayingView_Previews: PreviewProvider {
 
 struct HandGestureShortcutPrimaryAction: ViewModifier {
     public func body(content: Content) -> some View {
+        #if XCODE1600
```

Comment:
That's a nice trick, but that also means that once you'll migrate to Xcode 16.1 betas for example, this condition will be false (as `XCODE_MAJOR_VERSION` will be `1601` and `XCODE1600` will now be undefined). Sure, you probably expect to migrate to the stable Xcode 16.0 once it's out before moving to Xcode 16.1 betas, and this `#if` conditional is only intended to be temporary and will be removed once migration to stable Xcode 16 can be done, but still easy to miss one of those when cleaning up, so…

I'm wondering if it wouldn't instead be possible/enough to use [the built-in compiler control directives](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/statements/#Compiler-Control-Statements) like `#if compiler(>=6.0)` instead.
 - On one hand that would allow this condition to stay true as we migrate to higher versions of Xcode even if you forget to clean up those `#if` after the migration is over
 - On the other hand, semantically speaking what one would really want to test is the version of the SDK used, not the version of the compiler… but since those are updated in tandem when they update Xcode versions… 🤷

Reply:
@AliSoftware `XCODE_MAJOR_VERSION` will return `XCODE1600` for 16.1. On 15.4 it returns `XCODE1500`, because it's the major version.

<img width="1418" alt="Screenshot 2024-08-21 at 10 50 40" src="https://github.com/user-attachments/assets/e604fd0e-6243-415b-9e38-99434fd47f9a">

That being said, I think your solution requires fewer changes and it's more elegant so I'll change to it! Thanks for sharing that!

Reply:
Ah interesting. I wonder why they return 1600 and not 16 then 🤷